test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java

```diff
@@ -329,70 +343,94 @@ public void testPyScikitKMeans() throws Exception {
         String clustObj1Name = "py_clusters1";
         ExpressionMatrix data = new ExpressionMatrix().withType("log-ratio").withScale("1.0")
                 .withData(getSampleMatrix());
-        WorkspaceClient wscl = getWsClient();
-        wscl.saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
+        List<Tuple11<Long, String, String, String, Long, String, Long, String, String, Long, Map<String, String>>> getInfoEKN = getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
                 new ObjectSaveData().withName(exprObjName).withType("KBaseFeatureValues.ExpressionMatrix")
-                .withData(new UObject(data)))));
-        impl.clusterKMeans(new ClusterKMeansParams().withInputData(testWsName + "/" + 
+                        .withData(new UObject(data)))));
+        String getId = impl.clusterKMeans(new ClusterKMeansParams().withInputData(testWsName + "/" +
                 exprObjName).withK(3L).withOutWorkspace(testWsName).withOutClustersetId(clustObj1Name),
                 token, getContext());
-        ObjectData res = wscl.getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
+        ObjectData res = getWsClient().getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
                 .withName(clustObj1Name))).get(0);
+
+        //test if returned id matches id returned by ws saveobjects
+        Assert.assertEquals(getInfoEKN.get(0).getE7()  + "/" + getInfoEKN.get(0).getE1() + "/" +  getInfoEKN.get(0).getE5(), getId);
+
         FeatureClusters clSet = res.getData().asClassInstance(FeatureClusters.class);
         checkKMeansForSample(clSet);
     }
-    
-    @Test
```

Comment:
Is this supposed to be deleted? That'll deactivate the test.

Reply:
thanks, fixed